src/rust_connection/fd_read_write.rs

```diff
@@ -268,43 +283,69 @@ pub trait ReadFD {
         }
         Ok(())
     }
+
+    /// Moves this reader into or out of nonblocking mode.
+    ///
+    /// In nonblocking mode, reading becomes nonblocking. This means that such calls immediately
+    /// return. If an immediate result is not possible, an error with kind
+    /// [`std::io::ErrorKind::WouldBlock`] is returned.
+    fn set_nonblocking(&mut self, nonblocking: bool) -> Result<()>;
 }
 
 /// Wraps a [`std::io::Read`] to implement the [`ReadFD`] trait.
 ///
 /// No file descriptors will be received. Attempts to read bytes are forwarded to the underlying
 /// reader.
-#[derive(Debug)]
-pub struct ReadFDWrapper<R: Read + std::fmt::Debug>(R);
+pub struct ReadFDWrapper<R: Read>(R);
+
+impl<W: Read + Debug> Debug for ReadFDWrapper<W> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        f.debug_tuple("ReadFDWrapper").field(&self.0).finish()
+    }
+}
 
-impl<R: Read + std::fmt::Debug> ReadFDWrapper<R> {
+impl<R: Read> ReadFDWrapper<R> {
     /// Create a new `ReadFDWrapper` for the given reader.
     pub fn new(read: R) -> Self {
         Self(read)
     }
 }
 
-impl<R: Read + std::fmt::Debug> ReadFD for ReadFDWrapper<R> {
+impl<R: Read> ReadFD for ReadFDWrapper<R> {
     fn read(&mut self, buf: &mut [u8], _fd_storage: &mut Vec<RawFdContainer>) -> Result<usize> {
         self.0.read(buf)
     }
 
     fn read_exact(&mut self, buf: &mut [u8], _fd_storage: &mut Vec<RawFdContainer>) -> Result<()> {
         self.0.read_exact(buf)
     }
+
+    fn set_nonblocking(&mut self, _nonblocking: bool) -> Result<()> {
+        unimplemented!()
+    }
 }
 
 /// A version of [`std::io::BufReader`] that supports receiving file descriptors.
-#[derive(Debug)]
-pub struct BufReadFD<R: ReadFD + std::fmt::Debug> {
+pub struct BufReadFD<R: ReadFD> {
     inner: R,
     buf: Box<[u8]>,
     // The following two variables describe the range of available data in `buf`
     start: usize,
     end: usize,
 }
 
-impl<R: ReadFD + std::fmt::Debug> BufReadFD<R> {
+impl<R: ReadFD + Debug> Debug for BufReadFD<R> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        f.debug_struct("BufReadFD")
+            .field("inner", &self.inner)
+            .field("buf", &self.buf)
+            .field("start", &self.start)
+            .field("end", &self.end)
+            .finish()
+    }
+}
```

Comment:
Why not `derive`?

Reply:
How can I derive this? I thought that that derive fails if some member does not implement `Debug` (but cannot check right now, will do later)

Reply:
Oh, you are right. I did not know that `#[derive(Debug)]` works like this.

Reply:
I did some history rewriting and got rid of this part of the change. Thanks for teaching me some Rust. :-)